doc/src/base/c.md

```diff
@@ -13,11 +13,11 @@ Base.unsafe_store!
 Base.unsafe_modify!
 Base.unsafe_replace!
 Base.unsafe_swap!
-Base.unsafe_copyto!{T}(::Ptr{T}, ::Ptr{T}, ::Any)
+Base.unsafe_copyto!(::Ptr{T}, ::Ptr{T}, ::Any) where T
```

Comment:
These changes seem reasonable/good/bugfixes, but I'm worried how breaking they might be to docsystem building across the ecosystem.

Reply:
Is there a way to figure this out? From some JuliaCon talks for example I get the impression that it is possible to test many or all packages with a modified Julia version and see what happens. If the documentation of a package would require changes, then this would result in an error during precompilation or when building the docs.

Reply:
There is nanosoldier, but that only runs testsuites which often do not include building docs.

Reply:
In Base, most changes occur in docstrings (as opposed to separate documentation files), and a missing change gives an error during precompilation. So with nanosoldier we might still get an idea of how big the problem really is.

Reply:
I would also have thought that this issue should be detected by Nanosoldier.
The only way to circumvent this possible incompatibility is if we decide to formally allow this syntax for functions. (But then, it should be documented somewhere.)
Then, it should also formally be defined show this syntax is interpreted. If we just say that in the documentation, `fun{X}(args)` is _exactly_ equivalent to `fun(args) where {X}`, then it is clear that everything in `X` is a type variable (with possible subtyping). Apart from the case in which `fun` is the name of a type/constructor, as then it could also be a concrete type (or some other bitstype). I previously thought that it is not easily possible for the macro-called function determining which is the case, but as I now think of it, I'm no longer so sure. For outer constructors, for sure when the macro is called, the type must already exist. And in order for the Julia compiler to accept `fun{X}` without `X` being in the `where` clause, `X` must also be well defined (in the context of the calling module, which is available to the macro). I don't know how it is when the docstring is attached to an inner constructor, though.
Then, in the specification of what `fun{X}(args)` means when `fun` is a constructor, should we allow mixing actual generic types (coming first?) with parametric types from the arguments (pretty clumsy IMHO)? Or just disallow this shorthand notation?